packages/ref/src/cmip_ref/datasets/obs4mips.py

```diff
@@ -0,0 +1,311 @@
+from __future__ import annotations
+
+import re
+import traceback
+from pathlib import Path
+from typing import Any
+
+import pandas as pd
+import xarray as xr
+from ecgtools import Builder
+from loguru import logger
+from sqlalchemy.orm import joinedload
+
+from cmip_ref.config import Config
+from cmip_ref.database import Database
+from cmip_ref.datasets.base import DatasetAdapter
+from cmip_ref.datasets.cmip6 import _parse_datetime
+from cmip_ref.datasets.utils import validate_path
+from cmip_ref.models.dataset import Dataset, OBS4MIPSDataset, OBS4MIPSFile
+from cmip_ref_core.exceptions import RefException
+
+
+def extract_attr_with_regex(
+    input_str: str, regex: str, strip_chars: str | None, ignore_case: bool
+) -> list[Any] | None:
+    """
+    Extract version information from attribute with regular expressions.
+    """
+    if ignore_case:
+        pattern = re.compile(regex, re.IGNORECASE)
+    else:
+        pattern = re.compile(regex)
+    match = re.findall(pattern, input_str)
+    if match:
+        matchstr = max(match, key=len)
+        match = matchstr.strip(strip_chars) if strip_chars else matchstr.strip()
+        return match
+    else:
+        return None
+
+
+def parse_obs4mips(file: str) -> dict[str, Any | None]:
+    """Parser for obs4mips"""
+    keys = sorted(
+        list(
+            {
+                "activity_id",
+                "frequency",
+                "grid",
+                "grid_label",
+                "institution_id",
+                "nominal_resolution",
+                "realm",
+                "product",
+                "source_id",
+                "source_type",
+                "variable_id",
+                "variant_label",
+            }
+        )
+    )
+
+    try:
+        with xr.open_dataset(file, chunks={}, use_cftime=True) as ds:
```

Comment:
```suggestion
	time_coder = xr.coders.CFDatetimeCoder(use_cftime=True)
        with xr.open_dataset(file, chunks={}, time_coder=time_coder) as ds:
```

use_cftime is deprecated (you might have saw warning messages when running the test suite